appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/Keyframes.kt

```diff
@@ -44,7 +46,8 @@ data class Keyframes<ModelState>(
         queue[it].targetState
     }
 
-    val segmentProgress = MutableStateFlow(progressFlow.value - currentIndex)
+    fun getSegmentProgress(segmentIndex: Int): Flow<Float> =
+        progressFlow.map { it.toSegmentProgress(segmentIndex) }.filter { it in 0f..1f }
```

Comment:
The `{ it in 0f..1f }` could (and probably should) be moved into `toSegmentProgress`, as it semantically already shouldn't go beyond `1f`.